tests/framework/artifacts.py

```diff
@@ -0,0 +1,146 @@
+# Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+"""Define classes for interacting with CI artifacts in s3."""
+
+import os
+import platform
+import re
+
+from shutil import copyfile
+from typing import List
+from os import path
+from pathlib import Path
+
+import boto3
+import botocore.client
+
+
+MICROVM_CONFIG_EXTENSION = ".json"
+MICROVM_KERNEL_EXTENSION = ".bin"
+MICROVM_DISK_EXTENSION = ".ext4"
+
+
+class Artifact:
+    """A generic artifact manipulation class."""
+
+    bucket = None
+    key = None
+    local_folder = None
+
+    def __init__(self, bucket, key, type="misc"):
```

Comment:
Could the artifact `type` be part of an enum, so that we explicitly allow only a predefined set of types?

Reply:
I think this is a good idea.